### PR TITLE
Use `output` to save client cert file locally

### PR DESCRIPTION
**Pull request by lukehinds**
Use `output` to save client cert file locally

This PR is based on top of #78

### CODE REVIEW

File: cmd/sign.go [hunk 1]
```diff
@@ -140,6 +141,15 @@ var signCmd = &cobra.Command{
 			return err
 		}
 		fmt.Println("Rekor entry successful. URL: ", tlogEntry)
+
+		if viper.IsSet("output") {
+			clientPEM, _ := pem.Decode([]byte(certResp.Payload))
```
Comment (bobcallaway) [inline cmd/sign.go:146]:
```suggestion
```

File: cmd/sign.go [hunk 2]
```diff
@@ -140,6 +141,15 @@ var signCmd = &cobra.Command{
...
+			certPEM := pem.EncodeToMemory(clientPEM)
```
Comment (bobcallaway) [inline cmd/sign.go:147]:
```suggestion
```

File: cmd/sign.go [hunk 3]
```diff
@@ -140,6 +141,15 @@ var signCmd = &cobra.Command{
...
+			err = ioutil.WriteFile(viper.GetString("output"), certPEM, 0600)
```
Comment (bobcallaway) [inline cmd/sign.go:148]:
```suggestion
			err = ioutil.WriteFile(viper.GetString("output"), signingCertPEM, 0600)
```

**Review by bobcallaway — COMMENTED** [review]
no need to redo the PEM decode/encode as it is done above.

File: pkg/cryptoutils/certificate.go [hunk 4]
```diff
@@ -45,11 +45,13 @@ func MarshalCertificateToPEM(cert *x509.Certificate) ([]byte, error) {
...
 	result := []*x509.Certificate{}
 	remaining := pemBytes
+
 	for {
-		if remaining == nil {
+		if len(remaining) == 0 {
 			break
 		}
 		certDer, restBytes := pem.Decode(remaining)
+
```
Comment (cpanato) [inline pkg/cryptoutils/certificate.go:54]: nit: maybe the extra line should be in line 53

```
if len(remaining) == 0 {
	break
}

certDer, restBytes := pem.Decode(remaining)
if certDer == nil {
	return nil, errors.New("error during PEM decoding")
}
```

**Review by cpanato — DISMISSED** [review]
lgtm, a non tiny comment

**Comment by lukehinds** [comment]
should be good now @cpanato / @bobcallaway , DCO failure is from accepting bobs commits in the gui.

**Review by bobcallaway — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**